src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -57,8 +57,12 @@ public PathCacheHolder(Dictionary<StringSegment, object> path, Dictionary<int, o
         private readonly Stack<MemoryStream> _cachedMemoryStreams = new Stack<MemoryStream>();
 
         private int _numberOfAllocatedStringsValues;
+        private int _allocatedStringValuesCount;
```

Comment:
Why keep it as a flag, we already have the count on the list, no?

Reply:
I need this, because the underlying list can be 32k, but last non-null LSV can be at earlier index after cleanup.

Reply:
There is the size of the buffer, and there is the number of valid items there.
All of which should be managed by the list itself, not externally.

Reply:
Done